gems/aws-sdk-core/lib/aws-sdk-core/rest/request/body.rb

```diff
@@ -45,20 +49,58 @@ def build_body(params)
             params[@rules[:payload]]
           elsif @rules[:payload]
             params = params[@rules[:payload]]
+            if xml_builder? &&
+               @rules.shape.member?(@rules[:payload_member].location_name)
+              # serializing payload member for rest-xml is as follows:
+              # 1. Use the member locationName if the member value doesn't match the member's name (default)
+              # 2. Use the value of the locationName on the member's target if present
+              # 3. Use the shape name of the member's target
+              update_payload_location_name
+            end
             serialize(@rules[:payload_member], params) if params
           else
             params = body_params(params)
             serialize(@rules, params) unless params.empty?
           end
         end
 
+        def update_payload_location_name
+          @rules[:payload_member].location_name =
+            @rules[:payload_member].shape['locationName'] ||
+            @rules[:payload_member].shape.name
+        end
+
+        def update_payload_content_type(req)
+          # headers could be already populated if specified on input shape
+          case @rules[:payload_member].shape
+          when BlobShape
+            req.headers['Content-Type'] ||= 'application/octet-stream'
+          when StringShape
+            req.headers['Content-Type'] ||= 'text/plain'
+          when UnionShape
+            req.headers['Content-Type'] ||= if xml_builder?
```

Comment:
Ditto on formatting, it's just easier to read the other way.

Reply:
I didn't realize I could do that so this is niceee